File: 3334-apple-redistribution-into-boxes/3334-apple-redistribution-into-boxes.py
class Solution:
    def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
        S = sum(apple)
        h=[-n for n in capacity]
        heapify(h)
        cnt=0
        while S>0:
            S+=heappop(h)
            cnt+=1
        return cnt

        

File: 3334-apple-redistribution-into-boxes/3334-apple-redistribution-into-boxes.cpp
class Solution {
public:
    int minimumBoxes(vector<int>& apple, vector<int>& capacity) {
        int s = accumulate(apple.begin(),apple.end(),0);
        int cnt=0;
        make_heap(capacity.begin(),capacity.end());
        while (s>0) {
            pop_heap(capacity.begin(),capacity.end());
            s-=capacity.back();
            capacity.pop_back();
            cnt++;
        }
        return cnt;
    }
};